src/Sylius/Bundle/CoreBundle/spec/Validator/Constraints/ProductImageVariantsBelongToOwnerValidatorSpec.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\CoreBundle\Validator\Constraints;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use PhpSpec\ObjectBehavior;
+use Sylius\Bundle\CoreBundle\Validator\Constraints\ProductImageVariantsBelongToOwner;
+use Sylius\Component\Core\Model\ProductImageInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidatorInterface;
+use Symfony\Component\Validator\Context\ExecutionContextInterface;
+
+final class ProductImageVariantsBelongToOwnerValidatorSpec extends ObjectBehavior
+{
+    function let(ExecutionContextInterface $executionContext): void
+    {
+        $this->initialize($executionContext);
+    }
+
+    function it_is_a_constraint_validator(): void
+    {
+        $this->shouldImplement(ConstraintValidatorInterface::class);
+    }
+
+    function it_throws_an_exception_if_value_is_not_product_image(): void
+    {
+        $this
+            ->shouldThrow(\InvalidArgumentException::class)
+            ->during('validate', [new \stdClass(), new ProductImageVariantsBelongToOwner()])
+        ;
+    }
+
+    function it_throws_an_exception_if_constraint_is_not_product_image_variants_belong_to_owner(
+        Constraint $constraint,
+        ProductImageInterface $image,
+    ): void {
+        $this
+            ->shouldThrow(\InvalidArgumentException::class)
+            ->during('validate', [$image, $constraint])
+        ;
+    }
+
+    function it_adds_a_violation_if_any_variant_does_not_belong_to_a_product_which_is_an_owner(
```

Comment:
Shouldn't there be a positive case in addition to this case, when variants match product, like `it_does_nothing_...`

Reply:
Indeed, I've added additional spec